model/atmosphere/diffusion/tests/diffusion/integration_tests/test_benchmark_diffusion.py

```diff
@@ -0,0 +1,271 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import functools
+
+import pytest
+import netCDF4 as nc4
+from pathlib import Path
+import numpy as np
+from icon4py.model.common.initialization.topography_initialization import topography_initialization
+import icon4py.model.common.dimension as dims
+import gt4py.next as gtx
+from icon4py.model.common.decomposition import definitions
+import icon4py.model.common.grid.states as grid_states
+from icon4py.model.atmosphere.diffusion import diffusion, diffusion_states
+
+from icon4py.model.common.grid import (
+    geometry_attributes as geometry_meta,
+    vertical as v_grid,
+    geometry,
+)
+from icon4py.model.common.interpolation import interpolation_attributes, interpolation_factory
+from icon4py.model.common.metrics import (
+    metrics_attributes,
+    metrics_factory,
+)
+from icon4py.model.common.utils import data_allocation as data_alloc, device_utils
+
+import icon4py.model.common.states.prognostic_state as prognostics
+
+from icon4py.model.common.grid import geometry as grid_geometry
+
+from ..fixtures import *
+
+
+def construct_dummy_decomposition_info(grid, backend) -> definitions.DecompositionInfo:
+    """A public helper function to construct a dummy decomposition info object for test cases
+    refactored from grid_utils.py"""
+
+    on_gpu = device_utils.is_cupy_device(backend)
+    xp = data_alloc.array_ns(on_gpu)
+
+    def _add_dimension(dim: gtx.Dimension):
+        indices = data_alloc.index_field(grid, dim, backend=backend)
+        owner_mask = xp.ones((grid.size[dim],), dtype=bool)
+        decomposition_info.with_dimension(dim, indices.ndarray, owner_mask)
+
+    decomposition_info = definitions.DecompositionInfo(klevels=grid.num_levels)
+    _add_dimension(dims.EdgeDim)
+    _add_dimension(dims.VertexDim)
+    _add_dimension(dims.CellDim)
+
+    return decomposition_info
+
+
+@pytest.fixture
```

Comment:
Question: are these fixtures local to this file or would it make sense to move them to a `fixtures.py` file to be reused by other tests?